src/database/schema/UserSchema.ts

```diff
@@ -263,7 +263,7 @@ class UserSchema implements BaseDbSchema {
   get = async (db: SQLiteDatabase, id: string): Promise<UserSchema> => {
     const [{rows}] = await db.executeSql('SELECT * FROM users WHERE id = ?', [id]);
     if (rows?.length === 0) return null;
-    return Promise.resolve(rows[0].raw().map(this.fromDatabaseObject));
+    return Promise.resolve(UserSchema.fromDatabaseObject(rows[0].raw()[0]));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation assumes that the SQL query will always return a single record. However, if there are multiple records with the same `id` in the database, this could lead to unexpected behavior as it only returns the first record. It's better to handle such cases explicitly.

```diff
-    return Promise.resolve(UserSchema.fromDatabaseObject(rows[0].raw()[0]));
+    const rawRows = rows[0].raw();
+    if (rawRows.length > 1) {
+        throw new Error(`Expected a single record for id ${id}, but found ${rawRows.length}`);
+    }
+    return Promise.resolve(UserSchema.fromDatabaseObject(rawRows[0]));
```
This change ensures that an error is thrown when multiple records are found for the same `id`, which can help in identifying data integrity issues early.


<!-- This is an auto-generated comment by OSS CodeRabbit -->